Sofa/framework/Type/src/sofa/type/trait/is_fixed_array.h

```diff
@@ -0,0 +1,65 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+#include <type_traits>
+
+namespace sofa::type::trait
+{
+
+/// Detect if a type T has iterator/const iterator function, operator[](size_t) and defines a static size
+template<typename T>
+struct is_fixed_array
```

Comment:
Quite verbose. My version of the trait: https://godbolt.org/z/WTv9aKhvs or a concept in C++20: https://godbolt.org/z/5bj6EcvoG

Reply:
c++20 concepts !